src/ex2/Uilogic.js

```diff
@@ -0,0 +1,444 @@
+      this.updateTask(task.id, TYPE_DEADLINE, task.name, new Date(input.value), task.isAnimated)
+      input.nextElementSibling.innerHTML = new Date(input.value);
+    }
+    input.onclick = function () {
+      input.nextElementSibling.nextElementSibling.style.display = "block";
+    }
+    input.onblur = function () {
+      setTimeout(() => {
+        input.nextElementSibling.nextElementSibling.style.display = "none";
+      }, 500);
+    }
+    input.onmouseenter = function () {
+      if (input.nextElementSibling.innerHTML != "") {
+        input.nextElementSibling.style.display = "block";
+      }
+    }
+    input.onmouseleave = function (target) {
+      input.nextElementSibling.style.display = "none";
+    }
+    li.appendChild(input);
+
+    const div = document.createElement("div");
+    if (deadLineText) {
+      div.innerText = deadLineText;
+    }
+    div.classList.add("popup");
+    div.style.display = "none";
+
+    li.appendChild(div);
+    const btn = document.createElement("button");
+    btn.innerText = "Select Date"
+    btn.style.display = "none";
+    btn.classList.add('selectDate')
+
+    li.appendChild(btn);
+
+  }
+
+  formatDate(date) {
+    return (
+      [
+        date.getFullYear(),
+        this.padTo2Digits(date.getMonth() + 1),
+        this.padTo2Digits(date.getDate()),
+      ].join('-') +
+      ' ' +
+      [
+        this.padTo2Digits(date.getHours()),
+        this.padTo2Digits(date.getMinutes()),
+        // padTo2Digits(date.getSeconds()),  // 👈️ can also add seconds
+      ].join(':')
+    );
+  }
+  DisplayDropdown() {
+    document.getElementsByClassName("container")[0].style.visibility = "visible"; //display dropdown
+  }
+  VanishDropdown() {
+    document.getElementsByClassName("container")[0].style.visibility = "hidden"; //vanish dropdown
+  }
+
+
+  addPokemonImage(pokemonObj) {
+    const url = pokemonObj.sprites.front_default;
+    document.getElementById('imageNBA').setAttribute('src', url);
+    document.getElementsByClassName('NbaImage')[0].style.visibility = "visible"
+    const spinning = [
+      { transform: 'rotate(0deg) scale(0)' },
+      { transform: 'rotate(0) scale(1)' }
+
+    ];
+    const timing = {
+      duration: 2000,
+      iterations: 1,
+    }
+    document.getElementById('imageNBA').animate(spinning, timing);
+    setTimeout(() => {
+      const spinning = [
+        { transform: 'rotate(0) scale(1)' },
+        { transform: 'rotate(0deg) scale(0)' }
+      ];
+      const timing = {
+        duration: 2000,
+        iterations: 1,
+      }
+      document.getElementById('imageNBA').animate(spinning, timing);
+      setTimeout(() => { document.getElementsByClassName('NbaImage')[0].style.visibility = "hidden" }, 500)
+    }, 2000);
+  }
+
+  getPokemonObjectByName(taskList, taskObjectName) {
+    let pokemonObj = null;
+    const pokemonName = taskObjectName.split(' ');
+    taskList.forEach(task => {
+      if (task.isPokemon === true) {
```

Comment:
unnecessary comparison to true/false - fix everywhere
```suggestion
      if (task.isPokemon) {
```